lib/createGltf.js

```diff
@@ -13,335 +16,902 @@ module.exports = createGltf;
+    var baseColorImage = getImage(images, material.diffuseTexture, options.baseColorTexture, options);
+    var metallicImage = getImage(images, material.specularTexture, options.metallicRoughnessOcclusionTexture, options);
+    var roughnessImage = getImage(images, material.specularShininessTexture, options.metallicRoughnessOcclusionTexture, options);
+
+    var emissiveTexture = getTexture(gltf, emissiveImage);
+    var normalTexture = getTexture(gltf, normalImage);
+    var baseColorTexture = getTexture(gltf, baseColorImage);
+
+    var metallicRoughnessTexture;
+    if (defined(options.metallicRoughnessOcclusionTexture)) {
+        metallicRoughnessTexture = getTexture(gltf, metallicImage);
+    } else {
+        metallicRoughnessTexture = createMetallicRoughnessTexture(gltf, metallicImage, roughnessImage, occlusionImage, options);
+    }
+
+    var packOcclusion = (defined(occlusionImage) && options.packOcclusion) || defined(options.metallicRoughnessOcclusionTexture);
+    var occlusionTexture = packOcclusion ? metallicRoughnessTexture : getTexture(gltf, occlusionImage);
+
+    var emissiveFactor = getEmissiveFactor(material);
+    var baseColorFactor = material.diffuseColor;
+    var metallicFactor = material.specularColor[0];
+    var roughnessFactor = material.specularShininess;
+
+    if (defined(emissiveTexture)) {
+        emissiveFactor = [1.0, 1.0, 1.0];
+    }
+
+    if (defined(baseColorTexture)) {
+        baseColorFactor = [1.0, 1.0, 1.0, 1.0];
+    }
+
+    if (defined(metallicImage)) {
+        metallicFactor = 1.0;
+    }
+
+    if (defined(roughnessImage)) {
+        roughnessFactor = 1.0;
+    }
+
+    var alpha = material.alpha;
+    baseColorFactor[3] = alpha;
+
+    var transparent = alpha < 1.0;
+    if (defined(baseColorImage)) {
+        transparent = transparent || baseColorImage.transparent;
+    }
+
+    var doubleSided = transparent;
+    var alphaMode = transparent ? 'BLEND' : 'OPAQUE';
+
+    return {
+        name : materialName,
+        pbrMetallicRoughness : {
+            baseColorTexture : baseColorTexture,
+            metallicRoughnessTexture : metallicRoughnessTexture,
+            baseColorFactor : baseColorFactor,
+            metallicFactor : metallicFactor,
+            roughnessFactor : roughnessFactor
+        },
+        emissiveTexture : emissiveTexture,
+        normalTexture : normalTexture,
+        occlusionTexture : occlusionTexture,
+        emissiveFactor : emissiveFactor,
+        alphaMode : alphaMode,
+        doubleSided : doubleSided
+    };
+}
+
+function convertTraditionalToMetallicRoughness(material) {
+    // Translate the blinn-phong model to the pbr metallic-roughness model
+    // Roughness factor is a combination of specular intensity and shininess
+    // Metallic factor is 0.0
+    // This does not convert textures
+    var specularIntensity = material.specularColor[0];
```

Comment:
Is `specularColor` [r, g, b]? If so, this might warrant a comment explaining why `specularIntensity` is only based off one channel.